### DISCUSSION TITLE
Draft of guide for new storage providers

### DISCUSSION THREAD

**Discussion by de-authority** [filecoin-project/lotus#7317] [Tutorials]
## It's a draft of the Guide for new storage providers. any ideas/thoughts are welcome.

We see all sorts of questions in [slack](https://filecoin.io/slack) every day, and some of the basic questions have probably been answered a million times, So if we create a centralized document for new storage providers, we can help them join the filecoin network in the shortest possible time, so I've created this document to summarize all the materials and information, to give the community members a comprehensive understanding of our project, and to guide them on where to find resources and seek help when they encounter problems. If the following documents are linked, you can click on them directly to view the content, if they are not linked, it means that there are no suitable resources to link to, this document will be updated continuously and updated to [official docs](https://docs.filecoin.io/) if the content is complete

### start a small cluster

1. [start lotus daemon](https://github.com/filecoin-project/lotus/discussions/7524) 
2. [start lotus miner](https://github.com/filecoin-project/lotus/discussions/7547)
3. [start lotus worker](https://github.com/filecoin-project/lotus/discussions/7548)
4. [commit one sector and send one deal](https://github.com/filecoin-project/lotus/discussions/7571)

### Configuration file configuration reference

- lotus daemon configure option
- lotus miner configure option
- lotus worker configure option

##### run time environment variables
- lotus daemon run time environment variables
- lotus miner run time environment variables
- lotus worker run time environment variables

### examples of command and command options

- lotus
- lotus-worker
- lotus-miner

### more...

**Comment by stuberman** [comment]
Some suggestions for links:

Before you start:
Planning your disk space: https://github.com/filecoin-project/lotus/discussions/5989
Planning to buy hardware: https://github.com/filecoin-project/lotus/discussions/6071
Mining components (see diagram): https://filecoin.io/blog/posts/the-evolution-of-a-lotus-node/
Video drivers: https://github.com/filecoin-project/community/discussions/172

**Reply by de-authority** [reply]
Thank you very much @stuberman . really appreciate it!!!  I think this docs will make our new SP join filecoin easier. one of our criteria for choosing a document is that it is kept up to date, and you are really welcome to recommend such a document

**Comment by luckyz20** [comment]
Hi,boy！After start lotus miner, Here i have a question that what the pledge process, i had pledge a sector, can i pledge many sectors for a miner? "lotus-miner sectors mark-for-upgrade" what's the meaning of this? Help Please.

**Reply by de-authority** [reply]
Hi. i know you have many questions about how to start as a storage provider. but this disscussion is for how to supply docs to someone like you. not for specific questions.

**Reply by luckyz20** [reply]
okay

**Comment by luckyz20** [comment]
`lotus-miner proving info`

Miner: t017720
Current Epoch:           251393
Proving Period Boundary: 952
Proving Period Start:    251512 (in 59 minutes 30 seconds)
Next Period Start:       254392 (in 1 day 59 minutes)
Faults:      0 (0.00%)
Recovering:  0
Deadline Index:       39
Deadline Sectors:     0
Deadline Open:        253852 (in 20 hours 29 minutes)
Deadline Close:       253912 (in 20 hours 59 minutes)
Deadline Challenge:   253832 (in 20 hours 19 minutes)
Deadline FaultCutoff: 253782 (in 19 hours 54 minutes)

what is every arguments? i see official doc,but i don't quite understand，i hope than it's best to explain it in chinese, thinks

**Comment by luckyz20** [comment]
Just saw the mail.think you very much

chen wei ***@***.***>于2021年9月14日 周二11:41写道：

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment]
https://medium.com/encode-club/filecoin-club-how-mining-works-in-filecoin-video-slides-8fcea439107?source=friends_link&sk=1c7bb8cf954d4f338e240a25e35087ac what to understand before even thinking about hardware...

**Reply by de-authority** [reply]
Amazing Patrick.  I like this. BTW `Factor8.io` is in maintenance.  maybe you should bring it back.

**Reply by JelloHuang** [reply]
there are lack of some guide to deploy the lotus from scratch.so the work its valuable
